### ISSUE TITLE
Size

### ISSUE THREAD

**Issue by gtkpr** [Fusion/pngsource#4]
Was not sure where else to ask this, but it is unclear how big the files can be. In other words, how much source code can be embedded? What is the largest file size? What is the collective size of many files? Thanks

**Comment by Fusion** [comment]
Hi. Technically, you can embed one source code file per png file.
The maximum length of a PNG chunk is, theoretically, 2GB. You may have noticed that the chunk content is base64-encoded; base64 typically uses n * 3 / 4 space; this means that around 1.5 GB are available. Not bad (seriously!) but the story doesn't end here: the text being stored is compressed using the deflate algorithm. If your embedded text was English (with most DSLs requiring a considerably smaller dictionary) it would compress by a factor of roughly 3.

In conclusion, I have not tried it myself but you may be able to store about 4.5 GB of text. Note that the tool may end up using ungodly amounts of RAM if you attempt that! :) 

**Edit** I felt it was important to put this in perspective so here goes: *The Complete Works of William Shakespeare* is about 5.34MiB. It could therefore fit more than 582 times in a picture's text chunk.
In conclusion: the png specification is awesome.

**Comment by gtkpr** [comment]
@Fusion <3 Very cool!
